### ISSUE TITLE
The release workflow is not atomic nor idempotent

### ISSUE THREAD

**Issue by castarco** [nearform/optic-release-automation-action#86]
Right now, the release workflow is not atomic nor idempotent. It's possible that everything goes well for the most part, the npm package gets correctly published, and then the action ends up failing when it tries to create a "new release" in Github.

### Proposed solution:
In order to mitigate this problem, I propose to check if the npm package has been already published before even trying yo publish it. This way, if the previous described scenario happens in real life, we can re-run this action, and it will be able to skip the npm publication and directly jump into the GH release creation, allowing us to fix "unfinished" releases.

**Comment by simoneb** [comment]
I like the idea and propose another option, which I haven't validated though. We should run first the idempotent operations. If publishing the same release is idempotent, then we could swap the two operations and reduce the complexity of the logic we have in place. What I do not know is whether publishing an existing release is idempotent or not. Note that by publishing I mean, we have already created a draft release previously, we're just making it public

**Comment by castarco** [comment]
I just tried to re-publish one of my own packages, and it seems that npm doesn't like when we try to push the same stuff again:
```
npm ERR! code E403
npm ERR! 403 403 Forbidden - PUT https://registry.npmjs.org/@coderspirit%2fnominal - You cannot publish over the previously published versions: 3.2.1.
```
